src/http/websocket_http_client.zig

```diff
@@ -1020,6 +1032,9 @@ pub fn NewWebSocketClient(comptime ssl: bool) type {
             this.clearData();
             if (this.outgoing_websocket) |ws| {
                 this.outgoing_websocket = null;
+                const loop = JSC.VirtualMachine.get().eventLoop();
+                loop.enter();
+                defer loop.exit();
```

Comment:
(learning) So calling into the event `loop.enter()` and then to `exit()` after the block via `defer` is just to handle `ws.didAbruptClose(ErrorCode.ended);`?

Reply:
Yeah, internally we have a count if is 1 we free weak refs and drain microtasks